mir/src/node.rs

```diff
@@ -227,16 +228,86 @@ impl MirNode {
+                                                   c, self.name, self.columns);
+                                        },
+                                        None => {
+                                            let mut cn = c.clone().name;
+                                            match map.get(&cn) {
+                                                Some(ref mut res) => {
+                                                    let mut ind = 0;
+                                                    for self_column in &self.columns {
+                                                        if self_column.name == c.name && self_column.table == Some(res.clone()) {
+                                                            return ind;
+                                                        }
+                                                        ind = ind + 1;
+                                                    }
+                                                    panic!("tried to look up non-existent column {:?} on node \"{}\" (columns: {:?})",
+                                                           c, self.name, self.columns);
+                                                },
+                                                None => {
+                                                    panic!("alias is not in mapping table!");
+                                                }
+                                            }
+                                        }
+                                    }
+                                },
+                                None => {
+                                    let keyed = c.clone().name;
+                                    match map.get(&keyed) {
+                                        Some(ref mut t_n) => {
+                                            let mut ind = 0;
+                                            for self_column in &self.columns {
+                                                if self_column.name == c.name && self_column.table == Some(t_n.clone()) {
+                                                    return ind;
+                                                }
+                                                ind = ind + 1;
+                                            }
+                                            panic!("tried to look up non-existent column {:?} on node \"{}\" (columns: {:?})",
+                                                   c, self.name, self.columns);
+                                        },
+                                        None => {
+                                            panic!("tried to look up non-existent column {:?} on node \"{}\" (columns: {:?})",
+                                                   c, self.name, self.columns);
+                                        }
+                                    }
+                                }
+                            }
+                        },
```

Comment:
The above has a fair amount of repetition; I wonder if we could pull out the code that computes the column index and panics if the column doesn't exist, as it's identical except for the column name/table that we're looking for.
I'm thinking:
```rust
let needed_col = match *table_mapping { /* complex multi-case statement */ };
let ind = self.columns.position(|cc| cc = needed_col).expect("tried to ...");
```